### PR TITLE
Add changelog command to apidiff tool

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
The changelog command will generate tables for packages that are new,
updated, removed, and with breaking changes.  Table is generated in
markdown format.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] The PR targets the `latest` branch.
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
For https://github.com/Azure/azure-sdk-for-go/issues/931

File: tools/apidiff/cmd/changelog.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+func theChangelogCmd(args []string) error {
+	// there should be exactly three args, a directory and two commit hashes
+	if len(args) < 3 {
+		return errors.New("not enough args were supplied")
+	} else if len(args) > 3 {
+		return errors.New("too many args were supplied")
+	} else if strings.Index(args[2], ",") > -1 {
+		return errors.New("sequence of target commits is not supported")
+	}
+
```
Comment (marstr) [inline tools/apidiff/cmd/changelog.go]: I'd shove the logic above into the [Cobra `Args` property](https://godoc.org/github.com/spf13/cobra#Command).

File: tools/apidiff/cmd/changelog.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	rpt, err := thePackagesCmd(args)
```
Comment (marstr) [inline tools/apidiff/cmd/changelog.go]: I'm not crazy about this pattern, where commands end up calling each-other directly. It forces us to do argument checking in the body of this method, and it removes some strong typing that we could take advantage of in terms of what you need to execute.

Comment (jhendrixMSFT) [reply tools/apidiff/cmd/changelog.go]: Agreed.  I was in a bit of a hurry and didn't want to refactor the `packages` command a whole bunch.  :/  I will look at cleaning this up in a separate PR.

Comment (marstr) [reply tools/apidiff/cmd/changelog.go]: No worries, I totally get it. :)

**Pull request merged**